protoc-gen-gogo/generator/generator.go

```diff
@@ -3366,6 +3364,9 @@ func (g *Generator) generateMessage(message *Descriptor) {
 
 			oneofField := oFields[*field.OneofIndex]
 			tag = "protobuf:" + g.goTag(message, field, wiretype)
+			if gogoMoreTags != nil {
```

Comment:
I think we can just remove any tag logic here. 
I see the tags for every message does get constructed here:
https://github.com/gogo/protobuf/blob/23325ce401c6d7afd7008f26ff87c20074ea1413/protoc-gen-gogo/generator/generator.go#L2838-L2852

We just need to use that ```tag``` for our oneofSubField struct.